Dependencies/xr/Include/XR.h

```diff
@@ -131,12 +141,10 @@ namespace xr
                 std::unique_ptr<Impl> m_impl{};
             };
 
+            static arcana::task<std::shared_ptr<Session>, std::exception_ptr> CreateAsync(const Babylon::JsRuntimeScheduler& runtimeScheduler, System& system, void* graphicsDevice);
             Session(System& system, void* graphicsDevice);
             ~Session();
 
-            Session(Session&) = delete;
-            Session& operator=(Session&&) = delete;
-
```

Comment:
1.) Yes they should, but we discovered a bug in arcana where it would implicitly try to copy the ptr before executing the move, and we could not trick arcana into using the const ptr implementation of the get value on the expected.  We decided to just change this to shared_ptr for now as m_impl is already a unique_ptr and does all the real work, and investigate the arcana issue in the future.

2.) I would like to get rid off the public Session constructor, but this is implicitly needed to be public because make_shared enforces that you use a public constructor.

3.) These are implicitly deleted by the use of unique_ptr for the m_impl class.

Reply:
1) Sounds good to me. Can you add an issue to keep track of this, at least to this repo and perhaps to Arcana.cpp too? This workaround creates ownership ambiguity at the contract level, so let's just not lose track of it.

2) Makes sense.

3) That seems to be true for the copy constructor, but not for the move operator; from my quick tests, `struct Foo { std::unique_ptr<int> Bar{}; }` is still a moveable type. If we don't need the `Session` to be moveable, can we add back former line 138, or replace it with `Session(Session&&) = delete;`?

Reply:
We actually do need the session to be movable now to move the result passed back from CreateAsync to the NativeXR session, so I'm going to leave #3 as is.

I managed to get the constructor private by using the explicit std::shared_ptr function rather than make_share, so #2 should be fixed now.